src/org/zaproxy/zap/spider/URLCanonicalizer.java

```diff
@@ -96,17 +91,18 @@ private URLCanonicalizer(){}
 	 * @return the canonical url
 	 */
 	public static String getCanonicalURL(String url) {
-		return getCanonicalURL(url, null);
+		return getCanonicalURL(url, null, Collections.emptySet());
 	}
 
 	/**
 	 * Gets the canonical url, starting from a relative or absolute url found in a given context (baseURL).
 	 * 
 	 * @param url the url string defining the reference
 	 * @param baseURL the context in which this url was found
+	 * @param irrelevantParameters parameters that are skipped when canonicalizing
 	 * @return the canonical url
 	 */
-	public static String getCanonicalURL(String url, String baseURL) {
+	public static String getCanonicalURL(String url, String baseURL, Set<String> irrelevantParameters) {
```

Comment:
- It's probably better to use a functional interface (`Predicate<String>`?) instead of a passing a `Set`, it provides more flexibility.
- We should aim at keep binary compatibility (same for other public methods).

Reply:
Please clarify your point about passing predicate instead of a passing a Set, because I can't get how we can pass collection using functional interface that returns boolean.

Reply:
You pass the predicate, which makes use of the `Set`, e.g. `param -> mySet.contains(param)` (I'd expect the predicate to also make use of the `ExtensionHttpSessions` though).

Reply:
(Also, it can be shared by all `SpiderParser`s.)

Reply:
Yeah, great idea - I'll make changes. Thanks!